pkg/dialer/factory.go

```diff
@@ -152,24 +152,25 @@ func (f *Factory) clusterDialer(clusterName, address string) (dialer.Dialer, err
 			return cd(network, address)
 		}, nil
 	}
-	logrus.Debugf("dialerFactory: no tunnel session found for cluster [%s]", cluster.Name)
+	logrus.Debugf("dialerFactory: no tunnel session found for cluster [%s], falling back to nodeDialer", cluster.Name)
 
-	if cluster.Status.Driver != v3.ClusterDriverRKE {
-		return nil, fmt.Errorf("waiting for cluster agent to connect")
-	}
-
-	// Only for RKE will we try to connect to a node for the cluster dialer
+	// Try to connect to a node for the cluster dialer
 	nodes, err := f.nodeLister.List(cluster.Name, labels.Everything())
 	if err != nil {
 		return nil, err
 	}
 
+	var localAPIEndpoint bool
+	if cluster.Status.Driver == v3.ClusterDriverRKE {
+		localAPIEndpoint = true
+	}
+
 	for _, node := range nodes {
 		if node.DeletionTimestamp == nil && v3.NodeConditionProvisioned.IsTrue(node) {
 			logrus.Debugf("dialerFactory: using node [%s]/[%s] for nodeDialer", node.Labels["management.cattle.io/nodename"], node.Name)
 			if nodeDialer, err := f.nodeDialer(clusterName, node.Name); err == nil {
 				return func(network, address string) (net.Conn, error) {
-					if address == hostPort {
+					if address == hostPort && localAPIEndpoint {
```

Comment:
Limit override to 127.0.0.1 to rke clusters only, where every worker node has a reverse proxy that gives an access to the kube api